.github/workflows/diffcalc.yml

```diff
@@ -159,7 +170,6 @@ jobs:
     name: Setup environment
     needs: directory
     runs-on: self-hosted
-    if: ${{ !cancelled() && needs.directory.result == 'success' }}
```

Comment:
Did these conditions ever do anything at all? I'm asking because the removal partially looks like a serious change, partially looks like the conditions were simply bogus all along.

Reply:
I don't believe these are relevant anymore.

In a previous version, the ordering of the main graph was `create-comment --> ... --> generator --> ... --> update-comment`, so everything was dependent upon the `create-comment` job which is conditionally skipped. That's why it needed the `!cancelled()` check.

As to why that check couldn't have been `${{ success() }}`... There's weird semantics around what `success()` returns for skipped jobs and how it affects dependent jobs. Examples of others encountering this: [1](https://github.com/actions/runner/issues/2205) [2](https://github.com/orgs/community/discussions/45058) [3](https://stackoverflow.com/questions/74386773/github-actions-job-is-skipped-although-all-needs-succeeded)

But the `create-comment -> update-comment` branch has been isolated since, so I think it should be okay now.